### PR TITLE
fix some incorrect values + use latest kubectl when custom registry i…

### PR DESCRIPTION
**Pull request by haibinxie**
…s not specified

1. Some kubernetes versions do not have corresponding kubectl version, hence use latest when custom registry is not specified
2. fix some incorrect references to values.yaml

### CODE REVIEW

**Review by pure-jliao — APPROVED** [review]

File: community/portworx/templates/hooks/pre-upgrade/retain-daemonset-install.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ metadata:
   annotations:
     "helm.sh/hook": pre-upgrade
     "helm.sh/hook-weight": "10"
-    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+    "helm.sh/hook-delete-policy": before-hook-creation
```
Comment (piyush-nimbalkar) [inline community/portworx/templates/hooks/pre-upgrade/retain-daemonset-install.yaml]: Why not remove the hook on successful upgrade?

File: community/portworx/templates/_helpers.tpl [hunk 2]
```diff
@@ -20,6 +20,14 @@ productVersion: {{ .Values.imageVersion }}
 {{$version := .Capabilities.KubeVersion.GitVersion | regexFind "^v\\d+\\.\\d+\\.\\d+"}}{{$version}}
 {{- end -}}
 
+{{- define "px.getKubectlVersion" -}}
```
Comment (piyush-nimbalkar) [inline community/portworx/templates/_helpers.tpl:23]: Should we use the same function as the generic helm chart for consistency?
https://github.com/portworx/helm/blob/master/charts/portworx/templates/_helpers.tpl#L28

Comment (haibinxie) [reply community/portworx/templates/_helpers.tpl:23]: Sounds good, right now there are many differences in charts, a little painful to maintain.

Comment (haibinxie) [reply community/portworx/templates/_helpers.tpl:23]: Actually did a test, due to the kubectl image is from different docker repo, with that tag, the image is not found. I would keep as is to be safe.

File: community/portworx/templates/hooks/pre-upgrade/retain-daemonset-install.yaml (hunk 1 above)
Comment (haibinxie) [reply community/portworx/templates/hooks/pre-upgrade/retain-daemonset-install.yaml]: i removed it for debugging, added back

File: community/portworx/templates/_helpers.tpl (hunk 2 above)
Comment (piyush-nimbalkar) [reply community/portworx/templates/_helpers.tpl:23]: Got it. It is okay for now, but we might have issues with this in future. One such instance would be when APIs deprecate, then the newer kubectl versions may not recognize or worse try to manage objects considering the api server is updated too, which might not be the case. I believe we are not running any complicated commands using the kubectl in the hooks, so may it should be fine.

**Review by piyush-nimbalkar — APPROVED** [review]

**Pull request merged**